aioredis/commands/geo.py

```diff
@@ -0,0 +1,155 @@
+                  count=None, sort_dir=None, encoding=_NOTSET):
+        """Query a sorted set representing a geospatial index to fetch members
+        matching a given maximum distance from a point
+
+        :raises TypeError: radius is not float or int
+        :raises TypeError: count is not float or int
+        :raises ValueError: if sort not equal ASC or DESC
+        """
+        args = validate_georadius_options(
+            radius, unit, count, sort_dir, with_coord, with_dist, with_hash
+        )
+
+        fut = self._conn.execute(
+            b'GEORADIUS', key, longitude, latitude, radius,
+            unit, *args, encoding=encoding
+        )
+        return wait_convert_with_opts(
+            fut, geo_data_row,
+            with_coord=with_coord, with_dist=with_dist, with_hash=with_hash
+        )
+
+    def georadiusbymember(self, key, member, radius, unit='m',
+                          with_coord=False, with_dist=False, with_hash=False,
+                          count=None, sort_dir=None, encoding=_NOTSET):
+        """Query a sorted set representing a geospatial index to fetch members
+        matching a given maximum distance from a member
+
+        :raises TypeError: radius is not float or int
+        :raises TypeError: count is not float or int
+        :raises ValueError: if sort not equal ASC or DESC
+        """
+        args = validate_georadius_options(
+            radius, unit, count, sort_dir, with_coord, with_dist, with_hash
+        )
+        
+
+        fut = self._conn.execute(
+            b'GEORADIUSBYMEMBER', key, member, radius,
+            unit, *args, encoding=encoding
+        )
+        return wait_convert_with_opts(
+            fut, geo_data_row,
+            with_coord=with_coord, with_dist=with_dist, with_hash=with_hash
+        )
+
+
+def validate_georadius_options(radius, unit, count, sort_dir,
+                               with_coord, with_dist, with_hash):
+    args = []
+
+    if with_coord:
+        args.append(b'WITHCOORD')
+    if with_dist:
+        args.append(b'WITHDIST')
+    if with_hash:
+        args.append(b'WITHHASH')
+
+    if unit not in ['m', 'km', 'mi', 'ft']:
+        raise TypeError("unit argument must be 'm' or 'km' or 'mi' or 'ft'")
+    if not isinstance(radius, (int, float)):
+        raise TypeError("radius argument must be int or float")
+    if count:
+        if not isinstance(count, int):
+            raise TypeError("count argument must be int")
+        args += [b'COUNT', count]
+    if sort_dir:
+        if sort_dir not in ['ASC', 'DESC']:
+            raise ValueError("sort_dir argument must be euqal 'ASC' or 'DESC'")
+        args.append(sort_dir)
+    return args
+
+
+def pairs_float(value):
+    return [[float(val[0]), float(val[1])] for val in value]
+
+
+def geo_data_row(value, with_dist, with_coord, with_hash):
+    res_rows = []
+    for row in value:
+        res = []
+        
+        if with_dist and with_coord and with_hash:
```

Comment:
This looks overcomplicated. Lets make it simple.

First of all I think it would be nice to have these results wrapped in `namedtuple`
so users would not have to remember meaning of each index in a row:
```python
GeoCoord = namedtuple('GeoCoord', 'longtitude latitude')
GeoRadius = namedtuple('GeoRadius', 'name distance hash coord')

def make_geo_radius(name, distance=None, hash=None, coord=None):
    if distance is not None:
        distance = float(distance)
    if hash is not None:
        hash = int(hash)
    if coord is not None:
        hash = GeoCoord(*map(float, coord))
    return GeoRadius(name, distance, hash, coord)
```
So `geo_data_row` can look like this:
```python
...
for row in value:
    if isinstance(row, list):
        res_rows.append(make_geo_radius(*row))
    else:
        res_rows.append(row)
...
```